include/MouseAction.h

```diff
@@ -0,0 +1,25 @@
+#ifndef MOUSEACTION_H
+#define MOUSEACTION_H
+
+#include <QObject>
+
+class MouseAction : public QObject
+{
+    Q_OBJECT
+public:
+    virtual ~MouseAction();
+
+signals:
+    void Done(MouseAction* myself);
+
+protected:
+    QObject* m_watched;
+
+    MouseAction(QObject* watched);
```

Comment:
The constructor should have two parameters. One of type QObject* with the parent object (standard with Qt) and one of type QWidget* with the target widget. Of course, same is true for the derived classes.